### PR TITLE
fix: fix datasource result blocks

### PR DESCRIPTION
**Pull request by benPearce1**
related to #762

### CODE REVIEW

**Review by IsaacCalligeros95 — CHANGES_REQUESTED** [review]
Project Groups still needs updating. 

Variables => Scope still has a List Nested Block, I think this might be intentional and part of the actual config but taking a look would be good.

File: docs/data-sources/space.md [hunk 1]
```diff
@@ -15,16 +15,13 @@ Provides information about an existing space.
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
-### Required
-
-- `name` (String) The name of this resource, no more than 20 characters long
```
Comment (IsaacCalligeros95) [inline docs/data-sources/space.md:20]: This should still be required

File: octopusdeploy_framework/schemas/space.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func GetSpaceDatasourceSchema() map[string]datasourceSchema.Attribute {
 			Validators: []validator.String{
 				stringvalidator.LengthBetween(1, 20),
 			},
-			Required: true,
```
Comment (IsaacCalligeros95) [inline octopusdeploy_framework/schemas/space.go:103]: This is part of the individual space data source I think this should stay required.

File: octopusdeploy_framework/schemas/tag_set.go [hunk 3]
```diff
@@ -74,10 +74,38 @@ func GetTagSetDataSourceSchema() datasourceSchema.Schema {
 				Optional().
 				Description("A filter to specify the number of items to take (or return) in the response.").
 				Build(),
-			"tag_sets": datasourceSchema.ListAttribute{
+			"tag_sets": datasourceSchema.ListNestedAttribute{
 				Computed:    true,
```
Comment (IsaacCalligeros95) [inline octopusdeploy_framework/schemas/tag_set.go:78]: Are the `Optional: false` required? I didn't think so but if they are this is missing an Optional: false.

File: octopusdeploy_framework/schemas/space.go (hunk 2 above)
Comment (benPearce1) [reply octopusdeploy_framework/schemas/space.go:103]: Yes, this was a mistake

File: octopusdeploy_framework/schemas/tag_set.go (hunk 3 above)
Comment (benPearce1) [reply octopusdeploy_framework/schemas/tag_set.go:78]: I was just copying Matts original PR, will add Optional in here

File: docs/data-sources/space.md (hunk 1 above)
Comment (benPearce1) [reply docs/data-sources/space.md:20]: fixed

**Comment by benPearce1** [comment]
>Variables => Scope still has a List Nested Block, I think this might be intentional and part of the actual config but taking a look would be good.
@IsaacCalligeros95 I think when it comes to the datasource result set attributes, it probably doesn't matter. I think the resource schema types matter more

File: docs/data-sources/variables.md [hunk 4]
```diff
@@ -37,24 +36,11 @@ data "octopusdeploy_variables" "example" {
 - `is_editable` (Boolean) Indicates whether or not this variable is considered editable.
 - `is_sensitive` (Boolean) Indicates whether or not this resource is considered sensitive and should be kept secret.
 - `prompt` (Attributes List) (see [below for nested schema](#nestedatt--prompt))
+- `scope` (Attributes List) As variable names can appear more than once under different scopes, a VariableScope must also be provided (see [below for nested schema](#nestedatt--scope))
```
Comment (IsaacCalligeros95) [inline docs/data-sources/variables.md:39]: Looking at the old schema docs it looks like the scope should be Required, sorry I think I mislead you on the last review.

https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/isaac/v0.22.1-dsfix-copy/docs/data-sources/variables.md

File: docs/data-sources/tag_sets.md [hunk 5]
```diff
@@ -26,17 +26,17 @@ Provides information about existing tag sets.
...
-- `tag_sets` (List of Object) A list of tag sets that match the filter(s). (see [below for nested schema](#nestedatt--tag_sets))
+- `tag_sets` (Attributes List) A list of tag sets that match the filter(s). (see [below for nested schema](#nestedatt--tag_sets))
 
 <a id="nestedatt--tag_sets"></a>
 ### Nested Schema for `tag_sets`
 
-Read-Only:
+Optional:
 
-- `description` (String)
```
Comment (IsaacCalligeros95) [inline docs/data-sources/tag_sets.md:36]: I think these should still be Read only.

**Review by IsaacCalligeros95 — CHANGES_REQUESTED** [review]
A couple of minor comments.

File: docs/data-sources/variables.md (hunk 4 above)
Comment (benPearce1) [reply docs/data-sources/variables.md:39]: the scope is a nested property of the variable, within a datasource the results are all readonly, I think this feels right

**Review by IsaacCalligeros95 — APPROVED** [review]
LGTM

**Pull request merged**